### PR TITLE
Dashboards: Disable variable pickers for snapshots

### LINKED ISSUE #46097: Improve UX of variables in snapshots
Variables in snapshots show up as they'd do in a regular dashboard but they're actually not functional. This can be confusing for our users.

Proposed solution: disable variable input when they're in a snapshot

### PR DESCRIPTION
**Pull request by joshhunt**
**What this PR does / why we need it**:

Disables the variable picker UI when viewing a snapshot because you cannot change them anyway, and it's confusing for them to be interactable.

image

**Which issue(s) this PR fixes**:

Fixes #46097 

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/variables/adhoc/picker/ConditionSegment.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ interface Props {
 
 export const ConditionSegment: FC<Props> = ({ label }) => {
   return (
-    <div className="gf-form">
+    <div className="gf-form" data-wtf="condition-segment">
```
Comment (ivanortegaalba) [inline public/app/features/variables/adhoc/picker/ConditionSegment.tsx:9]: Just out of curiosity, what's this data attribute?

Comment (joshhunt) [reply public/app/features/variables/adhoc/picker/ConditionSegment.tsx:9]: oop my bad i should read my diffs before submitting them more 🙈

File: public/app/features/variables/pickers/shared/VariableLink.tsx [hunk 2]
```diff
@@ -88,28 +90,35 @@ const LoadingIndicator: FC<Pick<Props, 'onCancel'>> = ({ onCancel }) => {
...
-    border-radius: ${theme.border.radius.sm};
+    padding: 0 ${theme.spacing(1)};
+    background-color: ${theme.components.input.background};
+    border: 1px solid ${theme.components.input.borderColor};
+    border-radius: ${theme.shape.borderRadius(1)};
     display: flex;
     align-items: center;
     color: ${theme.colors.text};
-    height: ${theme.height.md}px;
+    height: ${theme.v1.height.md}px;
```
Comment (ivanortegaalba) [inline public/app/features/variables/pickers/shared/VariableLink.tsx:104]: Don't we have a replacement for `theme.v1` in theme v2?

**Review by ivanortegaalba — COMMENTED** [review]
Good job! 💯 While testing locally, I created every possible variable. 

I found:
* Input text get the hover styles when disabled
* Selects and input text doesn't display the `not-allowed` mouse pointer

File: public/app/features/variables/pickers/shared/VariableLink.tsx (hunk 2 above)
Comment (joshhunt) [reply public/app/features/variables/pickers/shared/VariableLink.tsx]: `GrafanaTheme2.v1` is like a compatibility shim that lets you opt back into the old theme for something. I initially thought that the old theme had a "better" expression of what we wanted, but you prompted to reevaluate and i updated it 😌

**Comment by lpskdl** [comment]
Thank you @joshhunt for the PR. I did the scenario we had during the mob session and everything looks good from my side 💯

**Comment by ivanortegaalba** [comment]
> Text box hover is a (reasonably) straightforward bug in the grafana-ui component we should fix, but would like to do it separately.

👍 

> The adhoc and text box components are "just" grafana-ui components, so any changes to them would need to be made in consideration to the whole system. This select picker _is_ custom to this part of the code, so we can freely make whatever changes we want.
> I think the 'best' solution is to just remove the `cursor: not-allowed` from grafana-ui Segment control (used for the adhoc picker) to make them all consistent, which we can do separately. Thoughts?

I'd like to check this with UX (not blocking this PR). If they agree any `disabled` component should have the `not-allowed` cursor, I'd try to push for that. IMHO, this is the best way to advise the user about the component state.

**Review by ivanortegaalba — APPROVED** [review]
All good! Thanks for applying the feedback! LGTM!

**Comment by fernsmark** [comment]
Allowing to pick the variables from filters is actually an important feature and it did work(v8.5.x) up until this release.

We still want to create a snapshot with the default 'All', the variable will still run the query and return the list of values to be selected. However, now in the UI it is not possible to make the selection. eg. the dashboard snapshot consists of many hosts and instead of showing All the hosts, we want to be able to filter by each host within the snapshot.

Please undo this change as it broke all the snapshots feature or provide an alternate solution.
Thanks.


grafana-snapshot-issue

**Pull request merged**